test/fakeintake/aggregator/common.go

```diff
@@ -50,8 +60,31 @@ func newAggregator[P PayloadItem](parse parseFunc[P]) Aggregator[P] {
 	}
 }
 
-// UnmarshallPayloads aggregate the payloads
+// UnmarshallPayloads parses a list of payloads and stores them in the aggregator
+// It replaces the current payloads with the new ones
 func (agg *Aggregator[P]) UnmarshallPayloads(payloads []api.Payload) error {
+	if *generateFixture {
+		_, filename, _, ok := runtime.Caller(0)
+		if !ok {
+			return errors.New("error generating fixture: unable to get the current file")
+		}
+
+		if len(payloads) == 0 {
+			return errors.New("error generating fixture: no payloads")
+		}
+
+		dir := filepath.Dir(filename)
+		// find aggregator name
+		tokens := strings.Split(reflect.TypeOf(agg.parse).Name(), "/")
+		aggName := tokens[len(tokens)-1]
+		aggName = strings.ReplaceAll(aggName, "]", "")
+		aggName = strings.ReplaceAll(aggName, "aggregator.", "")
+		aggName = strings.ReplaceAll(aggName, "Payload", "")
+		aggName = strings.ToLower(aggName)
+
+		return os.WriteFile(filepath.Join(dir, fmt.Sprintf("fixtures/%s_bytes", aggName)), payloads[0].Data, 0644)
```

Comment:
👏 **praise** :
That's very handy to save fixtures and implement other endpoints 👏